src/service/storage_arango.py

```diff
@@ -232,6 +242,104 @@ async def create_analyzer(
         """
         await self._db.create_analyzer(name, type_, properties, features)
     
+    async def create_search_view(
+        self,
+        name: str,
+        arango_collection: str,
+        view_spec: ColumnarAttributesSpec,
+        analyzer_provider: Callable[[FilterStrategy, bool], str]
+    ):
+        """
+        Create a search view for a collection.
+        
+        name - the name of the view to create.
+        arango_collection - the collection name for which to create the view.
+        view_spec - the specification for the view to create.
+        analyzer_provider - a function, that given a filter strategy, provides the name of
+            an analyzer to use for that strategy. The second argument defines whether to
+            return None (True) or the name of the default analyzer (False) when the default
+            analyzer is to be returned.
+        """
+        view_fields = self._view_spec_to_fields(view_spec, analyzer_provider)
+        try:
+            await self._db.create_arangosearch_view(
+                name, {"links": {arango_collection: {"fields": view_fields}}}
+            )
+        except ViewCreateError as e:
+            if e.error_code == ARANGO_ERR_NAME_EXISTS:
+                view = await self._db.view(name)
+                if view["links"][arango_collection]["fields"] != view_fields:
+                    raise ValueError(f"The view '{name}' already exists and differs from "
+                        + "the requested specification.") from e
+            else:
+                raise
+    
+    def _view_spec_to_fields(
```

Comment:
Just a general question here: 

You are not using async here because this function is mainly about data parsing/crunching?
Others you use async because they are mostly I/O bound task?

Reply:
>  If there's an await in the function, you have to make it async, if not you doni't (and shouldn't to avoid confusing readers).

Right, but how do you decide if very first function should be async or not?